t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -387,6 +387,55 @@ test_expect_success 'diff --staged' '
 	test_all_match git diff --staged
 '
 
+test_expect_success 'diff partially-staged' '
+	init_repos &&
+
+	write_script add-file-in-cone <<-\EOF &&
+	git sparse-checkout set deep
+	echo >deep/testfile
+	git add deep/testfile
+	echo a new line >>deep/testfile
+	EOF
+	run_on_all ../add-file-in-cone &&
+
+	test_all_match git diff &&
+	test_all_match git diff --staged &&
+
+	write_script add-file-outside-cone <<-\EOF &&
+	git reset --hard
+	mkdir newdirectory
+	echo >newdirectory/testfile
+	git sparse-checkout set newdirectory
+	git add newdirectory/testfile
+	echo a new line >>newdirectory/testfile
+	git sparse-checkout set
+	EOF
+	run_on_all ../add-file-outside-cone &&
+
+	test_all_match git diff &&
+	test_all_match git diff --staged &&
+
+	write_script merge-conflict-outside-cone <<-\EOF &&
+	git reset --hard
+	for side in left right
+	do
+		git checkout -b merge-$side base &&
+		git sparse-checkout set deep folder1 folder2
+		echo $side >>deep/deeper2/a &&
+		echo $side >>folder1/a &&
+		echo $side >>folder2/a &&
+		git add . &&
+		git commit -m "$side" || return 1
+		git sparse-checkout set
+	done
+	git merge merge-left
+	EOF
```

Comment:
This script is missing some `&&` and the loop body is a bit mixed up. Perhaps instead of doing this branch creation in this test, we could create the `merge-left`/`merge-right` branches inside the `setup` test? That makes it possible to re-use those branches for other tests.